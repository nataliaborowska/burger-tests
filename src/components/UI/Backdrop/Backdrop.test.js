import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import Backdrop from './Backdrop';

it('render correctly Backdrop component', () => {
  const tree = renderer.create(<Backdrop />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a div when props show is set to true', () => {
  const props = {
    show: true,
  };

  const BackdropComponent = shallow(<Backdrop {...props} />);
  const backdropNode = BackdropComponent.find('[data-test]');

  expect(backdropNode.length).toBe(1);
});

it('runs clicked prop when onClick', () => {
  const clicked = jest.fn();
  const props = {
    show: true,
    clicked: clicked,
  };

  const BackdropComponent = shallow(<Backdrop {...props} />);
  const backdropNode = BackdropComponent.find('[data-test]');
  backdropNode.simulate('click');

  expect(clicked).toHaveBeenCalledTimes(1);
});