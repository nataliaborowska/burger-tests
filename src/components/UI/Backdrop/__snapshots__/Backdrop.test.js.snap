// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Backdrop component 1`] = `null`;
